src/java.base/share/classes/java/lang/ref/Reference.java

```diff
@@ -403,13 +403,23 @@ boolean refersToImpl(T obj) {
      * necessary.
      */
     public void clear() {
+        clearImpl();
+    }
+
+    /* Implementation of clear(), overridden for phantom references.
+     * This method exists only to avoid making clear0() virtual. Making
+     * clear0() virtual has the undesirable effect of C2 often preferring
+     * to call the native implementation over the intrinsic.
+     */
+    void clearImpl() {
         clear0();
     }
 
     /* Implementation of clear(), also used by enqueue().  A simple
      * assignment of the referent field won't do for some garbage
      * collectors.
```

Comment:
Description of clear0 is rendered stale by this change.  The first sentence is no longer true, since it's now
clearImpl that has that role.  The second sentence probably ought to also be moved into the description of
clearImpl.

Reply:
Thanks! I tightened up comments a bit, take another look?

Reply:
Yes, that's better.  Thanks.